core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingUtils.java

```diff
@@ -52,6 +52,11 @@ public static void log(RequestContext ctx, RequestLog requestLog, LogWriter logW
                 logException(ctx, "response", t);
             }
         }
+        try {
+            logWriter.log(requestLog);
```

Comment:
What do you think of adding `logAtOnce` to `LoggingClient` and `LoggingService` instead of adding it to `logWriter` so that users can choose the way that they want?
```java
public static void log(RequestContext ctx, RequestLog requestLog, LogWriter logWriter, boolean logAtOnce) {
    if (logAtOnce) {
        try {
            logWriter.log(requestLog);
        } catch (Throwable t) {
            logException(ctx, "request and response", t);
        }
        return;
    }
    if (requestLog.requestCause() != null || !isTransientService(ctx)) {
        try {
            logWriter.logRequest(requestLog);
        } catch (Throwable t) {
            logException(ctx, "request", t);
        }
    }
    if (requestLog.responseCause() != null ||
        requestLog.responseHeaders().status().isServerError() ||
        !isTransientService(ctx)) {
        try {
            logWriter.logResponse(requestLog);
        } catch (Throwable t) {
            logException(ctx, "response", t);
        }
    }
}
```

Reply:
If it's acceptable, I think we'd better make a separate PR before working on this. 😉